### PR TITLE
chore: update tsc target

### PR DESCRIPTION
**Pull request by LeuisKen**
Update tsc target to "es2017" since vscode now using electorn 4.2.7
which based on chromium@68 and node@10.11.0.

This will enable more esnext features and reduce the bundle size.

According to: https://kangax.github.io/compat-table/es2016plus/

### CODE REVIEW

File: src/tsconfig.base.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
 		"module": "amd",
 		"moduleResolution": "node",
 		"noImplicitAny": true,
-		"target": "es5",
+		"target": "es2017",
```
Comment (jrieken) [inline src/tsconfig.base.json]: This shouldn't be needed given the change below

**Comment by jrieken** [comment]
Ran two builds to compare file sizes of our main bundle: before `6027569bytes`, after `5994648bytes`. That's less what I have expected but the awesome news is that async/await is now just as-is, no more yield, no more state machine.

**Comment by bpasero** [comment]
Why not go for ES2018, we actually seem to be using that already for all of our extensions, even though some seem to explicitly set a target:

https://github.com/microsoft/vscode/blob/20dd80d91a76cbe80fb1d1979c3a9f02d94ba161/extensions/shared.tsconfig.json#L3

I think:
* core should move to ES2018
* every extension should import the shared one
* no extension should overwrite the target

**Comment by LeuisKen** [comment]
I agree with @bpasero 's views.

**Comment by jrieken** [comment]
Using `es2018` yields in `5992860bytes` **but** it seems that dev tools fails to pretty print the minified file, e.g you must use source maps which isn't an option for everyone.

**Comment by LeuisKen** [comment]
> Using `es2018` yields in `5992860bytes` **but** it seems that dev tools fails to pretty print the minified file, e.g you must use source maps which isn't an option for everyone.

Yes, I also reproduce this problem on Chrome 76, so we can't fix this problem by using latest electron...
So what to do with this problem, shall I revert the "target" to "es2017" ?

File: src/tsconfig.json [hunk 2]
```diff
@@ -5,7 +5,7 @@
 		"preserveConstEnums": true,
 		"sourceMap": false,
 		"outDir": "../out",
-		"target": "es6",
+		"target": "es2018",
```
Comment (jrieken) [inline src/tsconfig.json]: Yes, back to `es2017`

File: extensions/markdown-language-features/preview-src/tsconfig.json [hunk 3]
```diff
@@ -1,13 +1,9 @@
...
 		"sourceMap": true,
-		"strict": true,
-		"strictBindCallApply": true,
-		"noImplicitAny": true,
-		"noUnusedLocals": true
+		"strictBindCallApply": true
 	}
-}
\ No newline at end of file
+}
```
Comment (jrieken) [inline extensions/markdown-language-features/preview-src/tsconfig.json]: These changes seem unrelated, please revert

**Review by jrieken — CHANGES_REQUESTED** [review]

**Review by jrieken — APPROVED** [review]

**Comment by bpasero** [comment]
@jrieken 

> Using es2018 yields in 5992860bytes but it seems that dev tools fails to pretty print the minified file, e.g you must use source maps which isn't an option for everyone.

To clarify: using ES2018 results in issues when looking at minified code? But we always ship with sourcemaps no?

**Comment by jrieken** [comment]
Well, when there is the need to debug minified code (because the issue is so obscure) then I don't trust source maps (they have failed me soo many times) and I want JS code.

**Comment by bpasero** [comment]
Yeah I saw this issue in action, weird. Even reproduces with latest E6 builds. Looks like a bug in DevTools to me.

**Comment by jrieken** [comment]
Yeah, sure an issue with dev tools. We can revisit this when we have a newer version or when we swap the minified.

**Comment by LeuisKen** [comment]
I create a issue for this: https://bugs.chromium.org/p/chromium/issues/detail?id=999794

**Comment by bpasero** [comment]
Thanks for doing that ❤️

**Comment by WSLUser** [comment]
you can switch to 2018 now. The bug says it's been fixed and has changed to using 2018.
